src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -95,6 +97,95 @@ CHIP_ERROR GeneralCommissioningAttrAccess::Read(const ConcreteReadAttributePath
+        Optional<TermsAndConditions> outTermsAndConditions;
+
+        if (nullptr == termsAndConditionsProvider)
+        {
+            return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+        }
+
+        CHIP_ERROR err = termsAndConditionsProvider->GetAcceptance(outTermsAndConditions);
+        if (CHIP_NO_ERROR != err)
+        {
+            return CHIP_ERROR_INTERNAL;
+        }
+
+        return aEncoder.Encode(outTermsAndConditions.ValueOr((TermsAndConditions){ .value = 0, .version = 0 }).version);
+    }
+    case TCMinRequiredVersion::Id: {
+        TermsAndConditionsProvider * const termsAndConditionsProvider = Server::GetInstance().GetTermsAndConditionsProvider();
+        Optional<TermsAndConditions> outTermsAndConditions;
+
+        if (nullptr == termsAndConditionsProvider)
+        {
+            return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+        }
+
+        CHIP_ERROR err = termsAndConditionsProvider->GetRequirements(outTermsAndConditions);
+        if (CHIP_NO_ERROR != err)
+        {
+            return CHIP_ERROR_INTERNAL;
+        }
+
+        return aEncoder.Encode(outTermsAndConditions.ValueOr((TermsAndConditions){ .value = 0, .version = 0 }).version);
+    }
+    case TCAcknowledgements::Id: {
+        TermsAndConditionsProvider * const termsAndConditionsProvider = Server::GetInstance().GetTermsAndConditionsProvider();
+        Optional<TermsAndConditions> outTermsAndConditions;
+
+        if (nullptr == termsAndConditionsProvider)
+        {
+            return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+        }
+
+        CHIP_ERROR err = termsAndConditionsProvider->GetAcceptance(outTermsAndConditions);
+        if (CHIP_NO_ERROR != err)
+        {
+            return CHIP_ERROR_INTERNAL;
+        }
+
+        return aEncoder.Encode(outTermsAndConditions.ValueOr((TermsAndConditions){ .value = 0, .version = 0 }).value);
+    }
+    case TCAcknowledgementsRequired::Id: {
+        TermsAndConditionsProvider * const termsAndConditionsProvider = Server::GetInstance().GetTermsAndConditionsProvider();
+        Optional<TermsAndConditions> outTermsAndConditions;
+        TermsAndConditionsState termsAndConditionsState;
+
+        if (nullptr == termsAndConditionsProvider)
+        {
+            return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+        }
+
+        CHIP_ERROR err = termsAndConditionsProvider->GetAcceptance(outTermsAndConditions);
+        if (CHIP_NO_ERROR != err)
+        {
+            return CHIP_ERROR_INTERNAL;
+        }
+
+        err = termsAndConditionsProvider->CheckAcceptance(outTermsAndConditions, termsAndConditionsState);
+        if (CHIP_NO_ERROR != err)
+        {
+            return CHIP_ERROR_INTERNAL;
+        }
+
+        bool setTermsAndConditionsCallRequiredBeforeCommissioningCompleteSuccess =
+            termsAndConditionsState != TermsAndConditionsState::OK;
+        return aEncoder.Encode(setTermsAndConditionsCallRequiredBeforeCommissioningCompleteSuccess);
```

Comment:
I understand the value read from this attribute change according to the user's status of acceptance.
However, it seems that there is no implementation to report the value when this attribute is subscribed.

Reply:
I'm updating the code to report attribute change for TCAcceptedVersion, TCAcknowledgements. TCAcknowledgementsRequired, when the state of the Read would have changes, based on SetTCAcknowledgements commands.

Reply:
Also, if any of these attributes are updated during runtime, maybe from some side-channel, then the manufacturer will need to extend or implement their own TermsAndConditionsProvider and make the appropriate call to `MatterReportingAttributeChangeCallback` when they change.